lib/validate.js

```diff
@@ -90,14 +93,21 @@ exports.usingBmap = (deviceFileDescriptor, options = {}) => {
  * });
  */
 exports.usingChecksum = (deviceFileDescriptor, options = {}) => {
+  const algorithms = options.imageChecksum
+    ? Object.keys(options.imageChecksum) : null;
+
   return checksum.calculateDeviceChecksum(deviceFileDescriptor, {
     imageSize: options.imageSize,
     progress: options.progress,
-    chunkSize: options.chunkSize
-  }).then((deviceChecksum) => {
-    if (options.imageChecksum !== deviceChecksum) {
-      throw errors.ValidationError;
-    }
+    chunkSize: options.chunkSize,
+    checksumAlgorithms: algorithms
+  }).then((results) => {
+
+    algorithms.forEach((algorithm) => {
+      if (options.imageChecksum[algorithm] !== results[algorithm]) {
```

Comment:
The opposite scenario might be worth testing: what happens if the user provides a checksum for an algorithm that was not passed to the algorithms list?